src/evergreen/evg_task_history.rs

```diff
@@ -35,6 +37,8 @@ pub struct HookRuntimeHistory {
     pub hook_name: String,
     /// Average runtime of hook.
     pub average_runtime: f64,
+    /// Max duration of test, in seconds
+    pub max_duration_pass: f64,
```

Comment:
* Renamed to `max_duration) in HookRuntimeHistory and TestRuntimeHistory.
* Re: Outside-of-expected values: That's a good question! Currently, the only scenario I'm handling is the zero-value case (zero indicates value-unavailable), in which case we fall back to the average runtime value. As for outlier detection, I definitely have nothing on that front (yet). As for splitting-tasks-based-on-sum, yep, that's the idea. But this would only be applied to tasks-where-we-care-about-max-time (notably, benchmark tests). WDYT?

Reply:
Here's where my head's at:
- TaskRuntimeHistory has many TestRuntimeHistory's.
- TestRuntimeHistory will have the max time spent on a test. A test can take up to 2 hours before timing out but typically may only take 2 minutes. So if evergreen is really slow one day it can jack up all the TestRuntimeHistory's "max times".
- If we sum all these "outlier max times" we get a massive outlier number that we then use to split into a massive number of tasks.

Right now, we purely use average time so even if we have these outliers, they have negligible impact on task splitting because they get averaged back out to -- say 2 minutes .

Reply:
Per offline conversation: I think we resolved this concern...? Since we're only going to leverage this feature for known-to-run-long benchmak tasks, it's ok. That is, we're using max time because we know that benchmarks run long, so we want a split-task calculation that uses the max times.